Taskfile.yml

```diff
@@ -8,23 +8,10 @@ tasks:
     desc: Template task to build a go binary
     dir: action
     cmds:
-      - GOOS={{.OS}} GOARCH=amd64 go build -ldflags="-s -w" -o ../bin/firmware-action-{{.OS}}-amd64-{{.SEMVER}}
-      - upx -9 ../bin/firmware-action-{{.OS}}-amd64-{{.SEMVER}} || true
-      # upx works, but always fails on darwin:
-      #   CantPackException: macOS is currently not supported (try --force-macos)
+      - go build -ldflags="-s -w" -o ../bin/firmware-action-{{OS}}-{{ARCH}}-{{.SEMVER}}
```

Comment:
While would always create a correct file name when compiling for the host target it might still create confusion when cross compiling in Go, given that `OS` and `Arch` are Taskfile specific variables that reflect the host calling the Taskfile.

Reply:
Yes, but I removed any option for cross-compilation from the taskfile. So that scenario should not happen.

I could go the other way around, keep cross-compilation, make this task `internal` and parametric and create instead tasks like `build-go-binary-linux` or `build-go-binary-windows` and so on.